### PR TITLE
Add load/store in DG

### PR DESCRIPTION
**Pull request by danbaterisna**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -589,6 +592,119 @@ Aspect: To allow updating of a similar object from User's list.
...
+Fig 22 - ExpenditureList Deserialization Sequence Diagram
+
+The above sequence diagram shows the interactions when a list of 
+`MMMSerializable`s is being deserialized. Although the given diagram shows the interaction for an 
+`ExpenditureList`, the interactions for `CreditCardList` and `IncomeList` are similar.
+
+1. `ExpenditureList` calls `SerializerFunctions.convertInputToList`, which accepts a function
+that deserializes a line of input. Here, `Expenditure.deserialize` is passed to `convertInputToList`.
+2. For each line in the input which corresponds to an `Expenditure`, `SerializerFunctions`
+invokes `Expenditure.deserialize` on this line, which returns an `Expenditure`. This is repeated until
```
Comment (khseah) [inline docs/DeveloperGuide.md:632]:
```suggestion
invokes `Expenditure.deserialize()` on this line, which returns an `Expenditure`. This is repeated until
```

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -589,6 +592,119 @@ Aspect: To allow updating of a similar object from User's list.
...
+the designated terminator is read from the `Scanner`.
+3. `Expenditure` invokes `PropertyList.deserialize` to convert the input line into a set of key-value pairs.
```
Comment (khseah) [inline docs/DeveloperGuide.md:634]:
```suggestion
3. `Expenditure` invokes `PropertyList.deserialize()` to convert the input line into a set of key-value pairs.
```

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -589,6 +592,119 @@ Aspect: To allow updating of a similar object from User's list.
...
+Then, it makes a series of `PropertyList::getValue` calls to obtain the values of each individual property.
```
Comment (khseah) [inline docs/DeveloperGuide.md:635]:
```suggestion
Then, it makes a series of `PropertyList.getValue()` calls to obtain the values of each individual property.
```

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -589,6 +592,119 @@ Aspect: To allow updating of a similar object from User's list.
...
+4. Once all attributes have been processed, `Expenditure.deserialize` returns an `Expenditure`.
```
Comment (khseah) [inline docs/DeveloperGuide.md:636]:
```suggestion
4. Once all attributes have been processed, `Expenditure.deserialize()` returns an `Expenditure`.
```

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -589,6 +592,119 @@ Aspect: To allow updating of a similar object from User's list.
...
+command execution.
+
+![Saving_sequence_diagram](images/Saving_Sequence_Diagram.png)
+<br/>
+
+Fig 23 - Saving Sequence Diagram
+
+The sequence diagram above shows the interactions when saving user data.
+
+1. After every command, `MMM` invokes `Storage::save`.
```
Comment (khseah) [inline docs/DeveloperGuide.md:652]:
```suggestion
1. After every command, `MMM` invokes `Storage.save()`.
```

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -589,6 +592,119 @@ Aspect: To allow updating of a similar object from User's list.
...
+2. `Storage` invokes `User::serialize`.
```
Comment (khseah) [inline docs/DeveloperGuide.md:653]:
```suggestion
2. `Storage` invokes `User.serialize()`.
```

File: docs/DeveloperGuide.md [hunk 7]
```diff
@@ -589,6 +592,119 @@ Aspect: To allow updating of a similar object from User's list.
...
+3. `User` calls `ExpenditureList::serialize`, which returns a `String`, representing the serialized
```
Comment (khseah) [inline docs/DeveloperGuide.md:654]:
```suggestion
3. `User` calls `ExpenditureList.serialize()`, which returns a `String`, representing the serialized 
```

File: docs/DeveloperGuide.md [hunk 8]
```diff
@@ -589,6 +592,119 @@ Aspect: To allow updating of a similar object from User's list.
...
+
+Fig 23 - Saving Sequence Diagram
+
+The sequence diagram above shows the interactions when saving user data.
+
+1. After every command, `MMM` invokes `Storage::save`.
+2. `Storage` invokes `User::serialize`.
+3. `User` calls `ExpenditureList::serialize`, which returns a `String`, representing the serialized 
+`ExpenditureList`.
+4. `User` then does the same for `CreditCardList::deserializeFrom` and `IncomeList::deserializeFrom`, which
```
Comment (khseah) [inline docs/DeveloperGuide.md:656]:
```suggestion
4. `User` then does the same for `CreditCardList.deserializeFrom()` and `IncomeList.deserializeFrom()`, which
```

File: docs/DeveloperGuide.md [hunk 9]
```diff
@@ -589,6 +592,119 @@ Aspect: To allow updating of a similar object from User's list.
...
+5. `User` compiles all these into one `String`, and returns this to `Storage`.
+6. `Storage` writes the returned serialized `User` into the data file.
+
+![Serialize_list_sequence_diagram](images/SerializeListSequenceDiagram.png)
+<br/>
+
+Fig 24 - ExpenditureList Serialization Sequence Diagram
+
+The above sequence diagram shows the interactions when a list of
+`MMMSerializable`s is being serialized. Although the given diagram shows the interaction for an
```
Comment (khseah) [inline docs/DeveloperGuide.md:667]: Is the 's' a typo?
```suggestion
`MMMSerializable` is being serialized. Although the given diagram shows the interaction for an
```

File: docs/DeveloperGuide.md [hunk 10]
```diff
@@ -589,6 +592,119 @@ Aspect: To allow updating of a similar object from User's list.
...
+`ExpenditureList`, the interactions for `CreditCardList` and `IncomeList` are similar.
+
+1. `ExpenditureList` calls `SerializerFunctions.addListToStringBuilder`, passing in an `ArrayList` of
```
Comment (khseah) [inline docs/DeveloperGuide.md:670]:
```suggestion
1. `ExpenditureList` calls `SerializerFunctions.addListToStringBuilder()`, passing in an `ArrayList` of 
```

File: docs/DeveloperGuide.md [hunk 11]
```diff
@@ -589,6 +592,119 @@ Aspect: To allow updating of a similar object from User's list.
...
+Fig 24 - ExpenditureList Serialization Sequence Diagram
+
+The above sequence diagram shows the interactions when a list of
+`MMMSerializable`s is being serialized. Although the given diagram shows the interaction for an
+`ExpenditureList`, the interactions for `CreditCardList` and `IncomeList` are similar.
+
+1. `ExpenditureList` calls `SerializerFunctions.addListToStringBuilder`, passing in an `ArrayList` of 
+`Expenditure`s.
+2. For each entry in the list, `SerializerFunctions`
+   invokes `Expenditure.propetyList` on this line, which returns an `Expenditure`. This is repeated until
```
Comment (khseah) [inline docs/DeveloperGuide.md:673]:
```suggestion
   invokes `Expenditure.propetyList()` on this line, which returns an `Expenditure`. This is repeated until
```

File: docs/DeveloperGuide.md [hunk 12]
```diff
@@ -589,6 +592,119 @@ Aspect: To allow updating of a similar object from User's list.
...
+   the designated terminator is read from the `Scanner`.
+3. `Expenditure` creates a `PropertyList` to stores its properties into.
+   Then, it makes a series of `PropertyList::setValue` calls to save each of its attributes.
```
Comment (khseah) [inline docs/DeveloperGuide.md:676]:
```suggestion
   Then, it makes a series of `PropertyList.setValue()` calls to save each of its attributes.
```

File: docs/DeveloperGuide.md [hunk 13]
```diff
@@ -589,6 +592,119 @@ Aspect: To allow updating of a similar object from User's list.
...
+4. Once all attributes have been processed, `PropertyList::serialize` is called, and
```
Comment (khseah) [inline docs/DeveloperGuide.md:677]:
```suggestion
4. Once all attributes have been processed, `PropertyList.serialize()` is called, and
```

File: docs/DeveloperGuide.md [hunk 14]
```diff
@@ -589,6 +592,119 @@ Aspect: To allow updating of a similar object from User's list.
...
+this serialized `PropertyList` is returned by `Expenditure::serialize`.
```
Comment (khseah) [inline docs/DeveloperGuide.md:678]:
```suggestion
this serialized `PropertyList` is returned by `Expenditure.serialize()`.
```

File: docs/puml/SerializeListSequenceDiagram.puml [hunk 15]
```diff
@@ -0,0 +1,35 @@
...
+    Expenditure -> PropertyList : serialize()
+    activate PropertyList STORAGE_COLOUR
+    PropertyList --> Expenditure --
+
+    Expenditure --> SerializerFunctions --
+end
+SerializerFunctions --> ExpenditureList --
+[<-- ExpenditureList --
+
+@enduml
```
Comment (khseah) [inline docs/puml/SerializeListSequenceDiagram.puml:35]:
```suggestion
@enduml

```

File: docs/puml/SavingSequenceDiagram.puml [hunk 16]
```diff
@@ -0,0 +1,32 @@
...
+activate CreditCardList USERFINANCIAL_COLOUR
+
+CreditCardList --> User --
+User -> IncomeList : serialize()
+activate IncomeList USERFINANCIAL_COLOUR
+
+IncomeList --> User --
+User --> Storage --
+deactivate Storage
+@enduml
```
Comment (khseah) [inline docs/puml/SavingSequenceDiagram.puml:32]:
```suggestion
@enduml

```

File: docs/puml/LoadingSequenceDiagram.puml [hunk 17]
```diff
@@ -0,0 +1,33 @@
...
+ExpenditureList --> User --
+User -> CreditCardList ** : deserializeFrom()
+activate CreditCardList USERFINANCIAL_COLOUR
+CreditCardList --> User --
+User -> IncomeList ** : deserializeFrom()
+activate IncomeList USERFINANCIAL_COLOUR
+IncomeList --> User --
+User --> Storage --
+Storage --> MMM --
+@enduml
```
Comment (khseah) [inline docs/puml/LoadingSequenceDiagram.puml:33]:
```suggestion
@enduml

```

File: docs/puml/DeserializeListSequenceDiagram.puml [hunk 18]
```diff
@@ -0,0 +1,33 @@
...
+    Expenditure -> PropertyList: getValue()
+    activate PropertyList STORAGE_COLOUR
+    PropertyList --> Expenditure --
+
+    Expenditure --> SerializerFunctions --
+end
+SerializerFunctions --> ExpenditureList --
+[<-- ExpenditureList --
+
+@enduml
```
Comment (khseah) [inline docs/puml/DeserializeListSequenceDiagram.puml:33]: Just add a newline so GitHub won't complain haha
```suggestion
@enduml

```

File: docs/puml/DeserializeListSequenceDiagram.puml [hunk 19]
```diff
@@ -0,0 +1,33 @@
+@startuml
+!include Style.puml
+
+box Deserialize List
+participant ":ExpenditureList" as ExpenditureList USERFINANCIAL_COLOUR
+participant ":SerializerFunctions" as SerializerFunctions STORAGE_COLOUR
+participant ":Expenditure" as Expenditure USERFINANCIAL_COLOUR
+participant ":PropertyList" as PropertyList STORAGE_COLOUR
+end box
+
```
Comment (khseah) [inline docs/puml/DeserializeListSequenceDiagram.puml:10]: Hide footbox to disable participants at the bottom of sequence diagram
```suggestion
hide footbox

```

File: docs/puml/LoadingSequenceDiagram.puml [hunk 20]
```diff
@@ -0,0 +1,33 @@
...
+
+box Loading
+participant ":MMM" as MMM MMM_COLOUR
+participant ":Storage" as Storage STORAGE_COLOUR
+participant ":User" as User USERFINANCIAL_COLOUR
+participant ":ExpenditureList" as ExpenditureList USERFINANCIAL_COLOUR
+participant ":CreditCardList" as CreditCardList USERFINANCIAL_COLOUR
+participant ":IncomeList" as IncomeList USERFINANCIAL_COLOUR
+end box
+
```
Comment (khseah) [inline docs/puml/LoadingSequenceDiagram.puml:12]:
```suggestion
hide footbox

```

File: docs/puml/SavingSequenceDiagram.puml [hunk 21]
```diff
@@ -0,0 +1,32 @@
...
+
+box Saving
+participant ":MMM" as MMM MMM_COLOUR
+participant ":Storage" as Storage STORAGE_COLOUR
+participant ":User" as User USERFINANCIAL_COLOUR
+participant ":ExpenditureList" as ExpenditureList USERFINANCIAL_COLOUR
+participant ":CreditCardList" as CreditCardList USERFINANCIAL_COLOUR
+participant ":IncomeList" as IncomeList USERFINANCIAL_COLOUR
+end box
+
```
Comment (khseah) [inline docs/puml/SavingSequenceDiagram.puml:12]:
```suggestion
hide footbox

```

File: docs/puml/SerializeListSequenceDiagram.puml [hunk 22]
```diff
@@ -0,0 +1,35 @@
+@startuml
+!include Style.puml
+
+box Serialize List
+participant ":ExpenditureList" as ExpenditureList USERFINANCIAL_COLOUR
+participant ":SerializerFunctions" as SerializerFunctions STORAGE_COLOUR
+participant ":Expenditure" as Expenditure USERFINANCIAL_COLOUR
+participant ":PropertyList" as PropertyList STORAGE_COLOUR
+end box
+
```
Comment (khseah) [inline docs/puml/SerializeListSequenceDiagram.puml:10]:
```suggestion
hide footbox

```

**Review by SeanHoWB — COMMENTED** [review]
looks good to me

**Pull request merged**